home/modules/contribute/pages/generate-rest-api.adoc

```diff
@@ -0,0 +1,242 @@
+│       └── index.adoc
+├── override <3>
+│   └── asciidoc
+│       ├── index.mustache
+│       ├── model.mustache
+│       ├── param.mustache
+│       ├── params.mustache
+│       └── stubs
+│           └── empty.adoc
+└── specs <4>
+    ├── rest-api-admin.yaml
+    ├── rest-api-metrics.yaml
+    └── rest-api-public.yaml
+----
+
+<1> builds the contents of `generated/`
+<2> contains partials for inclusion by Antora
+<3> contains mustache templates copied from Capella Core above, but with tweaks detailed below.
+<4> contains the 3 specs for the APIs (Public, Admin, and Metrics)
+
+The Generate script uses Docker to run openapi-generator - 
+this is convenient as you don't have to worry about installing the toolchain, but is a bit slower.
+
+[NOTE]
+--
+
+Some of the mustache files are changed in the following ways:
+
+. Change from e.g. `{{message}}` to `{{{message}}}` to prevent HTML escaping
+. surround in `[markdown]` block (see below)
+. where in a table, this requires prefixing the cell declaration as `a|`
+. replace e.g. `{{{notes}}}` with `{{{unescapedNotes}}}`
+. this unlovable workaround for lack of an `if` statement:
++
+[source,mustache]
+----
+{{#unescapedDescription?}}
+{{{unescapedDescription}}}
+{{/unescapedDescription?}}
+{{^unescapedDescription?}}
+{{{description}}}
+{{/unescapedDescription?}}
+----
+. Other minor edits
+--
+
+OpenAPI content can include Markdown, however openapi-generator 
+https://github.com/OpenAPITools/openapi-generator/issues/7765[does not convert Markdown]
+so we modify the mustache files above to wrap their output in a `[markdown]` block macro.
+
+This is defined in https://github.com/couchbase/docs-site/blob/HEAD/lib/markdown-block.js
+and should be included in your antora-playbook.yml to preview locally.
+
+
+
+
+==== Publishing the REST API
+
+Running the generate script above adds some partials to the source tree.
+You still need to publish the site in Antora.
+
+`docs/public/modules/app-services/pages/references/` contains three .adoc files for the API pages:
+
+. `rest_api_admin_static.adoc`
+. `rest_api_metrics_static.adoc`
+. `rest_api_public_static.adoc`
+
+These contain a bare skeleton and include the partials generated above.
+They declare some variables and `:page-role: openapi` which improves the formatting of the openapi tables, preventing excessive wrapping and hyphenation.
+(This requires docs-ui 149 or later).
+
+
+== Swagger UI
+
+You can include a Swagger UI block using the `swagger_ui::` block macro.
+See for example https://github.com/couchbase/docs-sdk-common/blob/release/7.1/modules/shared/partials/sample-application.adoc
+which is included by all the travel-sample app docs, for example https://docs.couchbase.com/java-sdk/current/hello-world/sample-application.html#rest-api
+
+
+== Swagger2Markup
+
+=== Couchbase Analytics
+
+NOTE: TODO, for SimonD
+
+=== Sync Gateway
+
+NOTE: Ian provided the following, which we couldn't get working, as the Swagger2Markup repositories
```

Comment:
Probably shouldn't mention team members by name. And again, this is sort of historical knowledge, so I don't think it needs to be included here.